src/rpc.rs

```diff
@@ -42,7 +42,7 @@ pub struct JsonRpcRequest<T> {
 
 impl<T> JsonRpcRequest<T> {
     pub fn new(method: &str, params: T) -> Self {
-        JsonRpcRequest {
+        Self {
```

Comment:
minor style q: do we have a preference as a team to use Self vs the type name itself?

Reply:
No need to enforce. I liked Self in this particular circumstance